Remove "docker engine" commands

thaJeztah opened the pull request:
relates to https://github.com/docker/cli/issues/2213

> These subcommands were created to allow upgrading a Docker Community engine to Docker Enterprise, but never really took off. With the migration of the docker "enterprise" products to Mirantis, we should remove these subcommands as releases may no longer be done in sync in future.

GordonTheTurtle commented:
Deploy preview for *docsdocker* ready!

Built with commit 564316de2c60b36e0f32929f683c21a6b8ccf01e

https://deploy-preview-9984--docsdocker.netlify.com

justincormack commented:
LGTM

thaJeztah commented:
rebased, and moved out of draft: the commands have not yet been removed from the `docker` CLI, but we won't be supporting this workflow, so we should remove the documentation.
